src/cached_object_store.rs

```diff
@@ -225,9 +228,9 @@ impl CachedObjectStore {
         Box::pin(async move {
             db_stats.object_store_cache_part_access.inc();
 
-            if let Ok(Some(bytes)) = entry.read_part(part_id).await {
+            if let Ok(Some(bytes)) = entry.read_part(part_id, range_in_part.clone()).await {
                 db_stats.object_store_cache_part_hits.inc();
-                return Ok(bytes.slice(range_in_part));
```

Comment:
I don't think it's a problem with the allocator's buffer management per se. I think that's happening is that we were returning the `Bytes` returned from `slice` here, which doesn't allocate a new buffer, but instead just adds a reference to the underlying buffer that stores the whole range and wraps that in the relevant offset+range. We then go and cache those Bytes in the in-memory cache (after slicing it a couple more times). So the moka cache thinks it has 4K blocks, but they are really 4K blocks that point to slices in 64MB ranges, and we blow up the memory usage from the in-memory cache. Anyway, all this is to say - I think the fix you've made here is sound, but I think we need to do something similar below where we have a cache miss and we fetch the object from the object store. Instead of returning `bytes.slice(range_in_part)`, we should instead copy that into a buffer of the right size.

Reply:
make sense :+1:

this explanation explains my confuse why this memory leaks only happens when both in memory cache and block storage cache are enabled.

Reply:
fixed another `bytes.slice()` in 1205b64dbee5a5f805dcc1219063cda05ee8bc2f